src/string.rs

```diff
@@ -62,11 +62,32 @@ pub fn rewrite_string<'a>(s: &str, fmt: &StringFormat<'a>) -> Option<String> {
         while !graphemes[cur_end - 1].trim().is_empty() {
             cur_end -= 1;
             if cur_end - cur_start < MIN_STRING {
-                // We can't break at whitespace, fall back to splitting
-                // anywhere that doesn't break an escape sequence.
                 cur_end = cur_start + max_chars;
-                while graphemes[cur_end - 1] == "\\" {
-                    cur_end -= 1;
+                //look for punctuation to break on
+                while graphemes[cur_end - 1] != ":" && graphemes[cur_end - 1] != "," &&
+                      graphemes[cur_end - 1] != ";" &&
+                      graphemes[cur_end - 1] != "." {
+                    if cur_end > 1 {
+                        cur_end -= 1;
+
+                    } else {
+                        break;
+                    }
+                }
+                // We can't break at whitespace or punctuation, fall back to splitting
+                // anywhere that doesn't break an escape sequence.
+                if cur_end < cur_start {
```

Comment:
You should compare to MIN_STRING here and in the `if` statement I said should become part of the predicate